### PR TITLE
Avoid handling mouseenter/mouseleave if device doesn't support hover

### LINKED ISSUE #11137: The header user menu requires two taps to open
Follow up for #11135 

### Describe the problem and steps to reproduce it:
1. Load AMO -dev on an android device
2. Log in
3. Tap on your user name in the page header to open the user menu

### What happened?
The user menu does not open at the the first tap, only at the second one

### What did you expect to happen?
The user menu opens at the first tap

studio64_Nfzt3vO7JK

### PR DESCRIPTION
**Pull request by diox**
This should fix the menu needing a double-tap to open on some mobile devices. I've tested it successfully on Desktop and Android, it fixes the issue for me without introducing regressions.

Fixes #11137

### CODE REVIEW

**Review by bobsilverberg — COMMENTED** [review]
This looks good. Regarding the test, you need to pass in a mock for the `_window` property in order to address the error you are seeing, and by configuring different mocks, you should be able to test the different logic paths that you have added.

For example, to create a mock that will result in `_window.matchMedia('(hover)').matches` evaluating to `true`, you would do: 
```
    const _window = {
      matchMedia: sinon.stub().returns({ matches: true }),
    };
```

and then, in the test, instead of `const root = mount(<DropdownMenu text="Menu" />);`, you would say `const root = mount(<DropdownMenu text="Menu" _window={_window} />);`.

For coverage you could create a mock where `...matches` returns `false`, and also one where `_window` is `undefined`.

**Comment by diox** [comment]
> and also one where `_window` is `undefined`.

This fails and I think this is expected? If I try to pass `undefined`, it will fall back to the default prop, which is `window` (no underscore), and we're back to needing `matchMedia`. Unless you meant passing `_window=undefined` *and* mocking `matchMedia` on the global `window` ?

**Comment by bobsilverberg** [comment]
> > and also one where `_window` is `undefined`.
> 
> This fails and I think this is expected? If I try to pass `undefined`, it will fall back to the default prop, which is `window` (no underscore), and we're back to needing `matchMedia`. Unless you meant passing `_window=undefined` _and_ mocking `matchMedia` on the global `window` ?

No, you're right and I was wrong. We cannot successfully test that scenario.

File: tests/unit/amo/components/TestDropdownMenu.js [hunk 1]
```diff
@@ -107,6 +111,42 @@ describe(__filename, () => {
...
+    getMenu().simulate('mouseEnter', createFakeEvent());
+    expect(getMenu()).not.toHaveClassName('DropdownMenu--active');
+
+    // User's mouse leaves the menu (no changes).
+    getMenu().simulate('mouseLeave', createFakeEvent());
+    expect(getMenu()).not.toHaveClassName('DropdownMenu--active');
+  });
+
+  it('does not touch active on mouseleave/enter if window is null', () => {
+    const _window = null;
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestDropdownMenu.js:134]: Nit: you could just inline this on line 138, but I think it's fine either way.

File: tests/unit/amo/components/TestDropdownMenu.js [hunk 2]
```diff
@@ -93,9 +93,13 @@ describe(__filename, () => {
   });
 
   it('sets active on mouseEnter/clears on mouseLeave', () => {
+    const _window = {
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestDropdownMenu.js:96]: If you want to add coverage that this was actually called, you could add an assertion about it in the test (the same goes for the next test), but I don't think that's necessary as we can see from the tests that it does in fact impact the behaviour of the component.

**Review by bobsilverberg — APPROVED** [review]
Looks good. I added a nit, and a comment about adding coverage, but I don't think we actually need it in this case.

r+wc

File: tests/unit/amo/components/TestDropdownMenu.js (hunk 2 above)
Comment (diox) [reply tests/unit/amo/components/TestDropdownMenu.js:96]: 50b07cc3d4839b71ecce6f9076387771a0b0d5bf just in case

**Pull request merged**